services/publish/src/main.rs

```diff
@@ -98,6 +132,31 @@ async fn async_main() -> anyhow::Result<(), anyhow::Error> {
     let addr: SocketAddr = format!("0.0.0.0:{}", port).parse()?;
     println!("Running on http://{}", addr);
 
+    // Background worker to clean up abandoned streams.
+    // Runs every WORKER_INTERVAL_SEC seconds and deletes streams with no activity for STREAM_IDLE_SEC+ seconds.
+    // Streams that reach Phase::End are untracked immediately since Redis TTL handles cleanup.
+    tokio::spawn(async move {
+        tracing::info!(
+            interval_sec = WORKER_INTERVAL_SEC,
+            idle_threshold_sec = STREAM_IDLE_SEC,
+            "Stream cleanup worker started",
+        );
+        let mut interval = tokio::time::interval(Duration::from_secs(WORKER_INTERVAL_SEC));
+        loop {
+            interval.tick().await;
+            let cutoff_timestamp = Utc::now().timestamp().sub(STREAM_IDLE_SEC);
+            let (total, deleted, untracked) =
+                match cleanup_old_stream(&broker, cutoff_timestamp).await {
+                    Ok(counts) => counts,
+                    Err(e) => {
+                        tracing::error!(error = %e, "Failed to get old streams for cleanup");
+                        continue;
+                    }
+                };
+            tracing::info!(total, deleted, untracked, "Cleanup cycle completed");
+        }
```

Comment:
How does this work get cleaned up gracefully? It seems like when a program shutdown signal is sent, this would be abruptly stopped. That could lead to your redis data structures in invalid states. Consider doing a clean shutdown to avoid these problems.